geomesa-kafka/geomesa-kafka-confluent/src/main/scala/org/locationtech/geomesa/kafka/confluent/ConfluentKafkaDataStore.scala

```diff
@@ -14,11 +14,16 @@ import io.confluent.kafka.schemaregistry.client.CachedSchemaRegistryClient
 import org.locationtech.geomesa.kafka.confluent.ConfluentGeoMessageSerializer.ConfluentGeoMessageSerializerFactory
 import org.locationtech.geomesa.kafka.data.KafkaDataStore
 import org.locationtech.geomesa.kafka.data.KafkaDataStore.KafkaDataStoreConfig
+import org.opengis.feature.simple.SimpleFeatureType
 
 object ConfluentKafkaDataStore {
+
   def apply(config: KafkaDataStoreConfig, schemaRegistryUrl: URL): KafkaDataStore = {
     val metadata = new ConfluentMetadata(new CachedSchemaRegistryClient(schemaRegistryUrl.toExternalForm, 100))
     val serialization = new ConfluentGeoMessageSerializerFactory(schemaRegistryUrl)
-    new KafkaDataStore(config, metadata, serialization)
+    new KafkaDataStore(config, metadata, serialization) {
+      override protected def preSchemaUpdate(sft: SimpleFeatureType, previous: SimpleFeatureType): Unit =
+        throw new NotImplementedError("Confluent Kafka stores do not support updateSchema")
```

Comment:
Why not?!;)

Reply:
The kafka confluent support reads from pre-defined topics/schemas, so if you want to add or modify them you would have to do it through the confluent schema registry.